### DISCUSSION TITLE
Simultaneous USB and internal camera ?

### DISCUSSION THREAD

**Discussion by andygodber** [Guilouz/Creality-Helper-Script-Wiki#581] [General]
Is it possible to run both the internal and USB camera with the USB Cam script?
Ive tried adding _1 to the stream but no joy.
I can get both working independantly, but thought it might operate like https://github.com/trevos3d/K1-USB-Cam

**Comment by Guilouz** [comment]
https://guilouz.github.io/Creality-Helper-Script-Wiki/helper-script/usb-camera-support/

**Reply by andygodber** [reply]
Thanks @Guilouz - just to clarify, I have installed the support and it works well (I can use the USB camera OR the internal camera) but I want to be able to use both at the same time?

**Reply by Guilouz** [reply]
Yes you can

**Reply by andygodber** [reply]
ah ok - my poor eyes didnt spot the webcam2 address ;) 
Now, to find out why that is not working.

**Reply by andygodber** [reply]
hmm, whatever I do, I cant seem to get the webcam2 content - always a message 502 Bad Gateway.
Anything I should look at for diagnostics?

**Reply by jlanton0000** [reply]
same here :-(

**Reply by Guilouz** [reply]
Are you sure your webcam is compatible ?

**Reply by andygodber** [reply]
> same here :-(

Please could you list your devices?

**Reply by jlanton0000** [reply]
root@K1Max-0E17 /root [#] lsusb
Bus 001 Device 008: ID a108:2231
Bus 001 Device 009: ID 095d:3001 Polycom, Inc.
Bus 001 Device 007: ID 1a86:55d3 QinHeng Electronics USB2.0 Hub
Bus 001 Device 006: ID 05e3:0610 Genesys Logic, Inc. 4-port hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

**Reply by jlanton0000** [reply]
from Mainsail UI

QINHENG ELECTRONICS
1A86:55D3
USB Single Serial
QinHeng Electronics
usb_location 1:7
serial 5764020960
class Communications

POLYCOM INC.
095D:3001
EagleEye Mini Camera
Polycom, Inc.
usb_location 1:9
serial 01.2025.4674
class Miscellaneous Device
protocol Interface Association

UNIONIMAGE CO.,LTD
A108:2231
CCX2F3299
Ingenic Semiconductor Co.,Ltd
usb_location 1:8
serial 1234567890
class Miscellaneous Device
protocol Interface Association

**Comment by andygodber** [comment]
> Are you sure your webcam is compatible ?

I believe so - it works if configured as the 'only' camera, and I successfully had both working using trevos3d approach.
![image](https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/1061812/ffdcc6b0-8b98-4146-b637-da3f64b94910)

**Reply by Guilouz** [reply]
I didn't see a second camera here, it must be there in /der/video6

**Comment by andygodber** [comment]
What is the device on /dev/video4 then? It certainly seems to disappear if I unplug it.... ?

**Comment by Supabad** [comment]
try adding 2 instead of 1 
![image](https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/485380/ab59444c-c50b-4da5-bbcd-7c31f59b13a6)

**Comment by Vaakefjell** [comment]
I'm having similar issues, and I can't get my head around it.

If I restart the printer, the built in camera streams OK, but there is no USB Camera. 

If I restart the /etc/init.d/S50usb_camera script, it switches to the USB Camera on the same stream (port 8080 or ...:4409/webcam/..., but there is nothing on port 8081 or ...:4409/webcam2/...). I tested with killing mpg_streamer and running the script commands manually, but the built in camera never starts.

Script commands:
/opt/bin/mjpg_streamer -b -i "/opt/lib/mjpg-streamer/input_uvc.so -d /dev/video4 -r 1280x720 -f 15" -o "/opt/lib/mjpg-streamer/output_http.so -p 8080"
/opt/bin/mjpg_streamer -b -i "/opt/lib/mjpg-streamer/input_uvc.so -d /dev/video6 -r 1280x720 -f 15" -o "/opt/lib/mjpg-streamer/output_http.so -p 8081"

**Reply by Vaakefjell** [reply]
Two minutes later I found the fix: The built in camera stream was on /dev/video5, not /dev/video6. Edit: /dev/video6 seem to be the h264 stream, while /dev/video5 is the YUV/MJPG stream, when running "v4l2-ctl -d /dev/video6 --list-formats-ext" 

Updated "/etc/init.d/S50usb_camera" to the following:

#!/bin/sh
# USB Camera Service
# Credit: destinal & Guilouz

case "$1" in
  start)
    echo "Starting USB Camera..."
    V4L_DEVS=$(v4l2-ctl --list-devices | grep -A1 usb | sed 's/^[[:space:]]*//g' | grep '^/dev')
    CREALITY_CAMS=$(v4l2-ctl --list-devices | grep -E 'CREALITY' | wc -l)
    if [ "x$V4L_DEVS" = "x" -o $CREALITY_CAMS -gt 0 ]; then
      /opt/bin/mjpg_streamer -b -i "/opt/lib/mjpg-streamer/input_uvc.so -d /dev/video5 -r 1280x720 -f 15" -o "/opt/lib/mjpg-streamer/output_http.so -p 8080"
      /opt/bin/mjpg_streamer -b -i "/opt/lib/mjpg-streamer/input_uvc.so -d /dev/video4 -r 1280x720 -f 15" -o "/opt/lib/mjpg-streamer/output_http.so -p 8081"
    else
      /opt/bin/mjpg_streamer -b -i "/opt/lib/mjpg-streamer/input_uvc.so -d /dev/video4 -r 1280x720 -f 15" -o "/opt/lib/mjpg-streamer/output_http.so -p 8080"
    fi
    ;;
  stop)
    echo "Stopping USB Camera..."
    killall -q mjpg_streamer && killall -q cam_app && killall -q cx_ai_middleware
    ;;
  restart|reload)
    "$0" stop
    sleep 1
    "$0" start
    ;;
  *)
    echo "Usage: /etc/init.d/S50usb_camera {start|stop|restart}"
    exit 1
esac

exit $?
